scripts/check.i18n.mjs

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env node
+
+import { readFileSync } from 'node:fs';
+import { join } from 'node:path';
+import { findFiles } from './utils.mjs';
+
+const PATH_FROM_ROOT = join(process.cwd(), 'src', 'frontend', 'src');
+const PATH_TO_EN_JSON = join(PATH_FROM_ROOT, 'lib', 'i18n', 'en.json');
+const PATH_TO_CODEBASE = join(PATH_FROM_ROOT);
+
+const extractKeys = (obj, prefix = '') =>
+	Object.keys(obj).reduce((res, el) => {
+		if (typeof obj[el] === 'object') {
+			return [...res, ...extractKeys(obj[el], `${prefix}${el}.`)];
+		}
+		return [...res, `${prefix}${el}`];
+	}, []);
+
+// It checks if the key is used in the content or if the key is used in a dynamic way
+const checkKeyUsage = ({ key, files }) =>
```

Comment:
We can maybe implement the same optimization as those we implemented in `check.unused.mjs` ? Keep a list of potential unused keys and remove those that matching and exist the loop as soon as there are no more entries?

Reply:
Good one! Done!